File: VB/PersonalOrganizer/ContactContextDataModel/IContactContextUnitOfWork.vb
Imports System
Imports System.Linq
Imports System.Data
Imports System.Data.Entity
Imports System.Linq.Expressions
Imports System.Collections.Generic
Imports PersonalOrganizer.Common.Utils
Imports PersonalOrganizer.Common.DataModel
Imports PersonalOrganizer.Common.DataModel.EntityFramework
Imports PersonalOrganizer.Model

Namespace PersonalOrganizer.ContactContextDataModel
    ''' <summary>
    ''' IContactContextUnitOfWork extends the IUnitOfWork interface with repositories representing specific entities.
    ''' </summary>
    Public Interface IContactContextUnitOfWork
        Inherits IUnitOfWork

        ''' <summary>
        ''' The Contact entities repository.
        ''' </summary>
        ReadOnly Property Contacts() As IRepository(Of Contact, Integer)
    End Interface
End Namespace

File: CS/PersonalOrganizer/ContactContextDataModel/IContactContextUnitOfWork.cs
using System;
using System.Linq;
using System.Data;
using System.Data.Entity;
using System.Linq.Expressions;
using System.Collections.Generic;
using PersonalOrganizer.Common.Utils;
using PersonalOrganizer.Common.DataModel;
using PersonalOrganizer.Common.DataModel.EntityFramework;
using PersonalOrganizer.Model;

namespace PersonalOrganizer.ContactContextDataModel {
    /// <summary>
    /// IContactContextUnitOfWork extends the IUnitOfWork interface with repositories representing specific entities.
    /// </summary>
    public interface IContactContextUnitOfWork : IUnitOfWork {

        /// <summary>
        /// The Contact entities repository.
        /// </summary>
        IRepository<Contact, int> Contacts { get; }
    }
}
